core/include/detray/tools/navigator.hpp

```diff
@@ -533,35 +523,14 @@ class navigator {
     }
 
     DETRAY_HOST_DEVICE
-    auto &get_detector() { return detector; }
+    auto &get_detector() { return _detector; }
```

Comment:
Does returning a reference to a const pointer makes much sense here?

Reply:
Ahh, good catch! That should really become `{ return *_detector; }` instead.

Aaaand... this is why I hate the over-use of `auto`. (Cough... @paulgessinger cough...)

Reply:
So... let's just make this into:

```c++
    DETRAY_HOST_DEVICE
    const detector_t &get_detector() { return _detector; }
```

Reply:
I guess ```auto``` is so widespread in detray, because we are still changing things a lot. We can remove them once the dust hast settled